website/files/models/base.py

```diff
@@ -91,6 +91,19 @@ def restore(self, recursive=True, parent=None):
         TrashedFileNode.remove_one(self)
         return restored
 
+    def get_guid(self, create=False):
+        """Attempt to find a Guid that points to this object.
+        One will be created if requested.
+        :rtype: Guid
+        """
+        try:
+            # Note sometimes multiple GUIDs can exist for
+            # a single object. Just go with the first one
+            return Guid.find(Q('referent', 'eq', self))[0]
+        except IndexError:
+            if not create:
+                return None
+        return Guid.generate(self)
```

Comment:
Unless I'm mistaken, this will generate a guid regardless of the ```create``` argument, right? Should this be in the ```except``` block?